save-backend/src/main/kotlin/com/saveourtool/save/backend/service/ExecutionService.kt

```diff
@@ -302,6 +280,46 @@ class ExecutionService(
         return savedExecution
     }
 
+    /**
+     * Delete [Execution] and links to TestSuite, TestExecution and related Agent with AgentStatus
+     *
+     * @param executionIds
+     */
+    @Transactional
+    fun deleteByIds(executionIds: List<Long>) {
+        log.info {
+            "Deleting executions with id in $executionIds. Additionally cleanup dependencies to these executions"
+        }
+        doDeleteDependencies(executionIds)
+        executionRepository.deleteAllById(executionIds)
+    }
+
+    /**
+     * Mark [Execution] as [ExecutionStatus.OBSOLETE] by ID
+     *
+     * @param executionIds
+     */
+    @Transactional
+    fun markAsObsoleteByIds(executionIds: List<Long>) {
+        log.info {
+            "Marking executions with id in $executionIds as obsolete. Additionally cleanup dependencies to these executions"
+        }
+        doDeleteDependencies(executionIds)
+        executionIds.forEach {
+            updateExecutionStatus(getExecution(it), ExecutionStatus.OBSOLETE)
+        }
+    }
+
+    private fun doDeleteDependencies(executionIds: List<Long>) {
```

Comment:
I don't think it's necessary. Tables have a cascade constraints. When deleting an execution, the data associated with it should be deleted from the rest of the tables.

Reply:
Agree, actually in one place we forget to remove LnkExecutionTestSuite and it was being cleanup by a cascade constrains.
But we still need to have it for obsolete, we can remove it for deleting an execution.

@petertrr @akuleshov, what's do you think about stopping deleting related entities on deleting and rely on database mechanism?

Reply:
I already had some problems with such delete, do you remember?

Reply:
> I already had some problems with such delete, do you remember?

We have a problem when execution's history was cleaned up after recreating (on fetching the same version) tests suite source. `Cascade constrains` removes invalid links

Reply:
> I do not remember what kind of problem, but we needed to rework delete at that time...

Current PR is just refactoring to avoid changes in #1670 when link to files will be added

Reply:
Will remove explicit deletion of dependencies on deletion of execution

Reply:
@Cheshiriks done

Reply:
added a note to https://github.com/saveourtool/save-cloud/issues/1575#issuecomment-1376928017